Ignore/Fix and Enable awsproviderlint AWSAT005 Hardcoded partition in ARN

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

> The [AWSAT005](https://github.com/terraform-providers/terraform-provider-aws/tree/master/awsproviderlint/passes/AWSAT005) analyzer reports hardcoded AWS partitions in ARNs. For tests to work across AWS partitions, the partitions should not be hardcoded.

Run the linter locally:

```console
% go install github.com/terraform-providers/terraform-provider-aws/awsproviderlint
go: finding module for package github.com/terraform-providers/terraform-provider-aws/awsproviderlint
go: downloading github.com/terraform-providers/terraform-provider-aws/awsproviderlint v0.0.0-20201014223354-b485d38053e4
go: found github.com/terraform-providers/terraform-provider-aws/awsproviderlint in github.com/terraform-providers/terraform-provider-aws/awsproviderlint v0.0.0-20201014223354-b485d38053e4
% awsproviderlint -AWSAT005 ./aws
/Volumes/develop/go/src/github.com/terraform-providers/terraform-provider-aws/aws/validators_test.go:2562:14: AWSAT005: avoid hardcoded ARN AWS partitions, use aws_partition data source
/Volumes/develop/go/src/github.com/terraform-providers/terraform-provider-aws/aws/validators_test.go:2570:14: AWSAT005: avoid hardcoded ARN AWS partitions, use aws_partition data source
...
```

### Existing Reports

- [x] cloudfront_distribution_configuration_structure_test.go (#15692)
- [x] data_source_aws_acmpca_certificate_authority_test.go (#15684)
- [x] data_source_aws_arn_test.go (#15684)
- [x] data_source_aws_cur_report_definition_test.go (#15684, #15724, #15883)
- [x] data_source_aws_elasticsearch_domain_test.go (#15684)
- [x] data_source_aws_iam_policy_document_test.go (#15665)
- [x] data_source_aws_lambda_invocation_test.go (#15684)
- [x] data_source_aws_secretsmanager_secret_test.go (#15684)
- [x] provider_test.go (#15664)
- [x] resource_aws_acmpca_certificate_authority_test.go (#15686)
- [x] resource_aws_alb_target_group_test.go (#15692)
- [x] resource_aws_cloudformation_stack_test.go (#15715)
- [x] resource_aws_cloudfront_origin_access_identity_test.go (#15715)
- [x] resource_aws_cloudtrail_test.go (#15663)
- [x] resource_aws_cloudwatch_event_target_test.go (#15666)
- [x] resource_aws_cloudwatch_log_resource_policy_test.go (#15717)
- [x] resource_aws_codedeploy_deployment_group_test.go (#15718)
- [x] resource_aws_cognito_identity_pool_test.go (#15720)
- [x] resource_aws_cognito_user_pool_client_test.go (#15720)
- [x] resource_aws_config_config_rule_test.go (#15667)
- [x] resource_aws_config_configuration_aggregator_test.go (#15667)
- [x] resource_aws_cur_report_definition_test.go (#15724)
- [x] resource_aws_db_instance_test.go (#15733)
- [x] resource_aws_db_option_group_test.go (#15733)
- [x] resource_aws_ecs_task_definition_test.go (#15679)
- [x] resource_aws_eks_cluster_test.go (#15649)
- [x] resource_aws_emr_cluster_test.go (#15586)
- [x] resource_aws_emr_instance_fleet_test.go (#15734)
- [x] resource_aws_emr_security_configuration_test.go (#15734)
- [x] resource_aws_flow_log_test.go (#15738)
- [x] resource_aws_gamelift_test.go (#15739)
- [x] resource_aws_glue_job_test.go (#15740)
- [x] resource_aws_glue_ml_transform_test.go (#15740)
- [x] resource_aws_guardduty_publishing_destination_test.go (#15741)
- [x] resource_aws_iam_service_linked_role_test.go (#15688)
- [x] resource_aws_iam_user_login_profile_test.go  (#15688)
- [x] resource_aws_iot_topic_rule_test.go (#15742)
- [x] resource_aws_kinesis_firehose_delivery_stream_migrate_test.go (#15677)
- [x] resource_aws_kinesis_firehose_delivery_stream_test.go (#15677)
- [x] resource_aws_kinesis_stream_migrate_test.go (#15677)
- [x] resource_aws_lambda_permission_test.go (#15713)
- [x] resource_aws_lb_listener_rule_test.go (#15673)
- [x] resource_aws_lb_target_group_test.go (#15673)
- [x] resource_aws_lb_test.go (#15673)
- [x] resource_aws_media_store_container_policy_test.go (#15743)
- [x] resource_aws_organizations_policy_test.go (#15744)
- [x] resource_aws_pinpoint_app_test.go (#15745)
- [x] resource_aws_route53_query_log_test.go (#15746)
- [x] resource_aws_s3_bucket_analytics_configuration_test.go (#15692)
- [x] resource_aws_s3_bucket_inventory_test.go (#15747)
- [x] resource_aws_sagemaker_model_test.go (#15678)
- [x] resource_aws_securityhub_product_subscription_test.go (#15670)
- [x] resource_aws_securityhub_standards_subscription_test.go (#15670)
- [x] resource_aws_shield_protection_test.go (#15750)
- [x] resource_aws_sns_platform_application_test.go (#15692, #15693)
- [x] resource_aws_sns_topic_test.go (#15693)
- [x] resource_aws_spot_fleet_request_test.go (#15693)
- [x] resource_aws_sqs_queue_test.go (#15693)
- [x] resource_aws_ssm_activation_test.go (#15693)
- [x] resource_aws_ssm_resource_data_sync_test.go (#15693)
- [x] resource_aws_storagegateway_cache_test.go (#15682)
- [x] resource_aws_storagegateway_cached_iscsi_volume_test.go (#15682)
- [x] resource_aws_storagegateway_upload_buffer_test.go (#15682)
- [x] resource_aws_storagegateway_working_storage_test.go (#15682)
- [x] resource_aws_transfer_user_test.go (#15748)
- [x] resource_aws_wafv2_web_acl_logging_configuration_test.go (#15749)
- [x] structure_test.go (#15692)
- [x] validators_test.go (#15692)
- [ ] Enable AWSAT005 (only after everything is merged) (#15751)

### Definition of Done

* Preferably fix or otherwise ignore existing reports
* `AWSAT005` check added to Makefile awsproviderlint target and CI passes

### References

* [awsproviderlint AWSAT005 Check](https://github.com/terraform-providers/terraform-provider-aws/tree/master/awsproviderlint/passes/AWSAT005)

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -97,6 +97,7 @@ awsproviderlint:
 		-AWSAT001 \
 		-AWSAT002 \
 		-AWSAT004 \
+		-AWSAT005 \
 		-AWSR001 \
 		-AWSR002 \
 		-R002 \

